app/validations/governing-body.js

```diff
@@ -0,0 +1,85 @@
+import { validatePresence } from 'ember-changeset-validations/validators';
+
+const governingBodyValidations = (governingBodies) => {
+  return {
+    startDate: [
+      validatePresence({
+        presence: true,
+        ignoreBlank: true,
+        message: 'Vul de startdatum in',
+      }),
+      validateStartDateBeforeEndDate(governingBodies),
+    ],
+    endDate: [
+      validatePresence({
+        presence: true,
+        ignoreBlank: true,
+        message: 'Vul de effectieve einddatum in',
+      }),
+      validateEndDateAfterStartDate(governingBodies),
+    ],
+  };
+};
+
+export default governingBodyValidations;
+
+function validateEndDateAfterStartDate(governingBodies) {
+  return (_key, newValue, oldValue, changes, content) => {
+    const endDateValue = newValue || oldValue;
+    const endDate = new Date(endDateValue);
+    const newStartDateValue = changes?.startDate || content.startDate;
+    if (newStartDateValue) {
+      const newStartDate = new Date(newStartDateValue);
+      if (newStartDate.getTime() > endDate.getTime()) {
+        return 'Kies een einddatum die na de startdatum plaatsvindt';
+      }
+      return validateNoOverlap(governingBodies, newStartDate, endDate);
+    }
+    return true;
+  };
+}
+function validateStartDateBeforeEndDate(governingBodies) {
+  return (_key, newValue, _oldValue, changes, content) => {
+    // The messages should also appear when the content of one of the fields is deleted.
+    const effectiveEndDateValue = changes?.endDate || content.endDate;
+
+    const newStartDate = new Date(newValue);
+
+    if (effectiveEndDateValue) {
+      const effectiveEndDate = new Date(effectiveEndDateValue);
+      if (newStartDate.getTime() > effectiveEndDate.getTime()) {
+        return 'Kies een startdatum die vóór de einddatum plaatsvindt';
+      }
+      return validateNoOverlap(governingBodies, newStartDate, effectiveEndDate);
+    }
+
+    return true;
+  };
+}
+
+function validateNoOverlap(otherBodies, startDate, endDate) {
```

Comment:
For the overlap, should we compare in strictly instead of `>=` / `<=` ? I'm thinking about all the organen we got from Loket, it's very common that one stops first of jan and the second one starts the same day. As they are datetimes, it's (I think) always at midnight, so it's probably safe to use strict comparison ?

Reply:
indeed! I've fixed it